model/api/src/main/java/org/jboss/aerogear/unifiedpush/api/IsAPNSVariant.java

```diff
@@ -0,0 +1,15 @@
+package org.jboss.aerogear.unifiedpush.api;
+
+import java.io.Serializable;
+
+/**
+ * There are two APNS Variants, iOSVariant and iOSTokenVariant.  However, they also have some metadata that is shared
+ * as well as needing to have their updates observed by SimpleApnsClientCache.  This is a interface that defines the
+ * common fields to update the cache as well as mark a variant as having APNS information.
+ */
+public interface IsAPNSVariant extends Serializable {
```

Comment:
I like the idea but the name of the interface is so confused looking into the code. Maybe we should just use `IAPNSVariant`, `APNSVariantType`, `APNSVariantInterface` or just `APNSVariant`.

@ziccardi Any suggestion for a better name?

Reply:
It is an interface implemented by all the iOS Variants, so why not simply `iOSVariantInterface` ?
Moreover: the `production` member variable and its setter and getter are duplicated into `iOSVariant` and `iOSTokenVariant`. The same for the `getType` method. I would suggest to replace the interface (or use together) with an abstract base class:

```suggestion
public abstract class AbstractIOSVariant implements Serializable {
   private final VariantType type;
   private boolean production;
   
   protected AbstractIOSVariant(final VariantType type) {
      this.type = type;
   }
   
   public getType() {
      return this.type;
   }
   
   public boolean isProduction() {
        return production;
   }

   public void setProduction(boolean production) {
        this.production = production;
   }
}
```

Reply:
Seems reasonable; the windows variants did the same thing.